influxql/engine.go

```diff
@@ -204,6 +204,50 @@ func (m *MapReduceJob) Execute(out chan *Row, filterEmptyResults bool) {
 	out <- row
 }
 
+// mergeOutputs merges two sorted slices of rawQueryMapOutput such that duplicate
```

Comment:
I don't follow -- how can we end up with duplicate timestamps? Data should only be de-duped if the series and timestamp is the same. But for a given series we should only be hitting 1 shard. I don't see how dupes can arise.